pleskdistup/common/src/dpkg.py

```diff
@@ -122,9 +126,54 @@ def safely_install_packages(
     install_packages(pkgs, repository, force_package_config)
 
 
+def apt_get_retry_temp_fails(
+    apt_get_cmd: typing.List[str],
+    tmpfail_retry_intervals: typing.Optional[typing.List[int]],
+    collect_stdout: bool = False
+) -> str:
+    cant_get_lock = False
+    stdout = []
+
+    if tmpfail_retry_intervals is None:
+        tmpfail_retry_intervals = DPKG_TEMPFAIL_RETRY
+
+    def process_stdout(line: str) -> None:
+        if collect_stdout:
+            nonlocal stdout
+            stdout.append(line)
+        log.info(line)
+
+    def process_stderr(line: str) -> None:
+        log.err(line)
+        nonlocal cant_get_lock
+        if cant_get_lock:
+            return
+        if "E: Could not get lock" in line:
+            cant_get_lock = True
+
+    i = 0
+    while True:
+        log.info(f"Executing: {' '.join(apt_get_cmd)}")
+        exit_code = util.exec_get_output_streamed(
+            apt_get_cmd, process_stdout, process_stderr,
+            env={"PATH": os.environ["PATH"], "DEBIAN_FRONTEND": "noninteractive"},
+        )
+        if exit_code == 0:
+            break
+        if i >= len(tmpfail_retry_intervals) or not cant_get_lock:
+            raise subprocess.CalledProcessError(returncode=exit_code, cmd=apt_get_cmd)
+        log.info(f"dist-upgrade failed because lock is already held, will retry in {tmpfail_retry_intervals[i]} seconds..")
```

Comment:
Why "dist-upgrade"? Did it really fail if we're going to retry?

Reply:
Thanks, it was wrong name